lib/starttls.js

```diff
@@ -33,111 +15,46 @@ function starttlsClient(socket, options, callback) {
 }
 
 function starttls(socket, options, callback, isServer) {
-  var sslcontext;
-
   var opts = {};
+
   Object.keys(options).forEach(function(key) {
     opts[key] = options[key];
   });
   if (!opts.ciphers) {
     opts.ciphers = RECOMMENDED_CIPHERS;
   }
+  opts.isServer = isServer;
+  opts.secureContext = tls.createSecureContext(opts);
 
-  socket.removeAllListeners('data');
-  if (tls.createSecureContext) {
-    sslcontext = tls.createSecureContext(opts);
-  } else {
-    sslcontext = crypto.createCredentials(opts);
-  }
-  var pair = tls.createSecurePair(sslcontext, isServer);
-  var cleartext = pipe(pair, socket);
-
+  var secureSocket = new tls.TLSSocket(socket, opts);
   var erroredOut = false;
-  pair.on('secure', function() {
+
+  // NodeJS documentation bug: secure vs secureConnect
+  // https://github.com/nodejs/node/issues/10555
+  secureSocket.on('secure', function() {
     if (erroredOut) {
-      pair.end();
+      secureSocket.end();
       return;
     }
 
-    var verifyError = (pair._ssl || pair.ssl).verifyError();
-
-    if (verifyError) {
-      cleartext.authorized = false;
-      cleartext.authorizationError = verifyError;
+    var authError = secureSocket.ssl.verifyError();
+    if (authError) {
+        secureSocket.authorized = false;
+        secureSocket.authorizationError = authError;
     } else {
-      cleartext.authorized = true;
+        secureSocket.authorized = true;
```

Comment:
I agree. By skipping server identity check we can make people to believe that the connection is fully secure while it is not. IMHO it is really a corner case as I doubt that any one runs FTP server configured to validate client certificates but nevertheless we can either:

 1) put a comment into the code explaining how the things are (as you suggest) or
 2) throw exception from FTP server constructor if anyone tries to use `requestCert` option to avoid false impression of security or
 3) allow user to specify his own checkServerIdentity method in options and throw exception if it is not specified (and requestCert is true)

@mk-pmb: What would be the preferred solution in your opinion? I'm open to other suggestions as well. thanks